addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -80,25 +89,30 @@
             ConfigConstants.getUserDataFolder() + File.separator + "hueemulation" + File.separator + "usernames");
     private static final File UDN_FILE = new File(
             ConfigConstants.getUserDataFolder() + File.separator + "hueemulation" + File.separator + "udn");
-
-    private static final String[] SUPPORTED_TAGS = new String[] { "Switchable", "Lighting", "TargetTemperature" };
-
+    private static final File ITEM_FILE = new File(
+            ConfigConstants.getUserDataFolder() + File.separator + "hueemulation" + File.separator + "items");
+    private static final String[] SUPPORTED_TAGS = new String[] { "Switchable", "Lighting" };
     private Gson gson = new Gson();
     private HttpService httpService;
     private ItemRegistry itemRegistry;
     private EventPublisher eventPublisher;
     private HueEmulationUpnpServer disco;
     private String udn;
     private String xmlDoc;
-    private CopyOnWriteArrayList<String> userNames = new CopyOnWriteArrayList<String>();
-
+    private int webPort;
     private boolean pairingEnabled = false;
+    // list of valid Hue API user ids
+    private ArrayList<String> userNames = new ArrayList<String>();
```

Comment:
I missed this one, next time use an interface for the declaration like `List `instead of the implementation, The same for line 107